src/contexts/auth-context.js

```diff
@@ -0,0 +1,95 @@
+import { createContext, useContext, useReducer, useState } from 'react';
+import { logInHandler, signUpHandler } from '../utils/auth-util';
+
+const authContext = createContext();
+const useAuth = () => useContext(authContext);
+
+const initialState = {
+  isActive: false,
+  showPassword: false,
+  email: '',
+  password: '',
+  rememberMe: false,
+  signUpFirstName: '',
+  signUpLastName: '',
+  signUpEmail: '',
+  signUpPassword: '',
+};
+
+const signInReducer = (state, action) => {
+  switch (action.type) {
+    case 'RESET_FORM':
+      return {
+        ...state,
+        showPassword: false,
+        email: '',
+        password: '',
+        rememberMe: false,
+        signUpFirstName: '',
+        signUpLastName: '',
+        signUpEmail: '',
+        signUpPassword: '',
```

Comment:
You can destructure initial state here rather than passing all the values again manually

Reply:
I have one state that I don't want to change that's why.